enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,442 @@
+| darwin/arm64 | arm64* |	cli-artifacts |	/usr/share/openshift/mac_arm64/oc | openshift-client-mac-arm64-%s.tar.gz |
+| windows/amd64 |	amd64* | cli-artifacts | /usr/share/openshift/windows/oc.exe | openshift-client-windows-%s.tar.gz |
+
+| Install platform | Image arch | Image |	Path | Tarball |
+| :--- | :--- | :--- | :--- | :--- |
+| linux/amd64 |	amd64 | installer |	/usr/bin/openshift-install | openshift-install-linux-amd64-%s.tar.gz |
+| linux/arm64	| arm64	| installer | /usr/bin/openshift-install | openshift-install-linux-arm64-%s.tar.gz |
+| linux/ppc64le |	ppc64le | installer | /usr/bin/openshift-install | openshift-install-linux-ppc64le-%s.tar.gz |
+| linux/s390x	| s390x	| installer	| /usr/bin/openshift-install | openshift-install-linux-s390x-%s.tar.gz |
+| darwin/amd64 | amd64* | installer-artifacts |	/usr/share/openshift/mac/openshift-install | openshift-install-mac-%s.tar.gz |
+| darwin/arm64 | arm64* | installer-artifacts | /usr/share/openshift/mac_arm64/openshift-install | openshift-install-mac-arm64-%s.tar.gz |
+
+*Technically it shouldn't matter which manifest's installer-artifacts are used, the windows and mac binaries are all cross-compiled anyway.
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for a graph for heterogeneous
+payloads which CVO will need to look at to make upgrade recommendations. There are two upgrade scenarios to consider with the addition of
+heterogeneous payloads:
+
+- heterogeneous -> heterogeneous release payload upgrade after a fresh install
+- homogeneous -> heterogeneous release payload upgrade
+
+A separate heterogeneous graph will be introduced for each version starting from the version which supports heterogeneous payloads. A cluster
+installed as a day1 heterogeneous cluster with a heterogeneous payload will use the heterogeneous graph for further updates.
+
+An existing homogeneous cluster can be converted to a heterogeneous cluster through a migration process which is a special upgrade process. The
+migration would be a separate lifecycle event triggered through a separate command `oc adm upgrade migrate-arch` (name TBD) which would:
+- identify the current cluster version, channel and architecture
+- fetch the heterogeneous graph and release corresponding to the version and channel
+- upgrade to the heterogeneous release
+- set the architecture to "multi" in the cluster version status for consumers to know the type of the cluster
+
+As part of this migration, the ClusterVersion API's spec would need to have an additional "architecture" field to indicate the architecture of the cluster.
+This field would indicate whether the cluster is a homogeneous or heterogeneous cluster with the individual arches denoting a homogeneous cluster and
+"multi" indicating a heterogeneous cluster. This field would also be part of the ClusterVersionStatus to indicate to consumers of the API who would want to
```

Comment:
is there going to be a metric exposing this so that we can slice our telemetry data based on this information?

Reply:
Yes in the future we do want to have this. Today for collecting data on the number of clusters per arch we have to look at a node's arch label. This will make it easier going forward to identify the cluster arch from the ClusterVersionStatus field rather than looking at a node's arch in that cluster.